### PR TITLE
feat(IDPay): [IODPAY-45] Added BonusCardScreen

### PR DESCRIPTION
**Pull request by forrest57**
## Short description
Added a BonusDetails screen, along with its new required card component and included it in the AppStackNavigator.

<img width="200" src="https://user-images.githubusercontent.com/60693085/203976392-9d2a2cde-b47e-4e69-b817-1ca5ca202afc.png"

## List of changes proposed in this pull request
- Implementation of the BonusDetailsScreen page.
- Addition of said page to the AppStackNavigator in order to be able to start the IDPAY workflow from anywhere in the app. 
- Introduction of the new BonusCardComponent, a refactor of BonusVacanze's card component which uses an SVG as background and flex values instead of hardcoded ones to accomodate the new data display requirements.
- addition of required sagas and reducers
- addition of IDPAY actions and reducers groupers 
- both the page and the component display conditional data as per design.

## How to test
using the token of a CF which completed an initiative's onboarding, navigate to the wallet to find the newly added initiative cards, then click any of those to display the page, if the initiative has not been configured yet, you will also find a CTA that allows navigation to the configuration flow.
Also, tests for the store and saga parts of the PR are included.

### CODE REVIEW

File: ts/navigation/params/AppParamsList.ts [hunk 1]
```diff
@@ -4,27 +4,28 @@ import {
...
 import { StackNavigationProp } from "@react-navigation/stack";
+import { CdcBonusRequestParamsList } from "../../features/bonus/cdc/navigation/params";
+import { CDC_ROUTES } from "../../features/bonus/cdc/navigation/routes";
 import {
   CgnActivationParamsList,
   CgnDetailsParamsList,
   CgnEYCAActivationParamsList
 } from "../../features/bonus/cgn/navigation/params";
 import CGN_ROUTES from "../../features/bonus/cgn/navigation/routes";
-import UADONATION_ROUTES from "../../features/uaDonations/navigation/routes";
```
Comment (forrest57) [inline ts/navigation/params/AppParamsList.ts:13]: import sorting, nothing was actually deleted

File: ts/features/idpay/initiative/details/components/BonusCardComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    borderRadius: 8,
+    borderTopWidth: 10,
+    borderTopColor: opaqueBorderColor,
+    height: 15,
+    width: "100%",
+    maxWidth: 343
+  }
+});
+
+const BonusCardComponent = (props: Props) => {
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/components/BonusCardComponent.tsx:73]:
```suggestion
const InitiativeCardComponent = (props: Props) => {
```

please also use the term Initaitive in the rest of the code.

File: ts/features/idpay/initiative/details/components/BonusCardComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const renderFullCard = () => {
+    // const maybeStatusDescription = maybeNotNullyString(
+    //   props
+    //     ? TypedI18n.t(`bonus.${props.status.toLowerCase()}`, {
+    //         defaultValue: ""
+    //       })
+    //     : ""
+    // );
+    const isBonusActive = props.status === StatusEnum.REFUNDABLE;
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/components/BonusCardComponent.tsx:82]:
```suggestion
    const isInitiativeConfigured = props.status === StatusEnum.REFUNDABLE;
```

File: ts/features/idpay/initiative/details/components/BonusCardComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+          <View style={styles.row}>
+            <View>
+              <Text
+                bold={true}
+                style={[
+                  !isBonusActive ? styles.consumedOpacity : {},
+                  styles.Amount
+                ]}
+              >
+                {`${props.amount},00 €`}
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/components/BonusCardComponent.tsx:121]: Use the proper amount formatter function.

File: ts/features/idpay/initiative/details/components/BonusCardComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          </View>
+        </View>
+        <View>
+          <View
+            style={styles.bottomBonusIcon}
+            accessible={true}
+            importantForAccessibility={"no-hide-descendants"}
+            accessibilityElementsHidden={true}
+          >
+            <Image source={bonusVacanzeWhiteLogo} style={styles.logo} />
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/components/BonusCardComponent.tsx:154]: Why bonus vancaze?

File: ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    position: "absolute",
+    alignSelf: "center",
+    width: widthPercentageToDP(90),
+    maxWidth: 343,
+    height: 192,
+    top: 2
+  }
+});
+
+const BonusDetailsList = (bonus: InitiativeDTO) => (
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx:35]: Why not a component?

Comment (francescopersico) [reply ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx:35]: Also why BonusDetailsList? This component is not a List of BonusDetail.

File: ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        subTitle={bonus.iban}
+      />
+    </List>
+  </>
+);
+
+export const BonusDetailsScreen = (
+  props: IOStackNavigationRouteProps<BonusDetailsScreenProps>
+) => {
+  const bonus: InitiativeDTO = {
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx:71]: Why you need this **inside** the component?

File: ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+            IOStyles.horizontalContentPadding,
+            { paddingTop: customVariables.contentPadding }
+          ]}
+        >
+          <View spacer extralarge />
+          <View spacer small />
+          {isNotRefundable ? (
+            <Text>INIZIATIVA NON CONFIGURATA, ID: {bonus.initiativeId}</Text>
+          ) : (
+            BonusDetailsList(bonus)
```
Comment (francescopersico) [inline ts/features/idpay/initiative/details/screens/BonusDetailsScreen.tsx:117]: Read above.

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/features/idpay/initiative/details/saga/__test__/handleGetInitiativeDetails.test.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+      handleGetInitiativeDetails,
+      getWallet,
+      mockToken,
+      mockLanguage,
+      { initiativeId: "123" }
+    )
+      .withReducer(appReducer)
+      .put(idpayInitiativeGet.failure(mockFailure))
+      .run();
+  });
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/saga/__test__/handleGetInitiativeDetails.test.ts:59]: Left is called only when there is an API exception. An API error (status code != 200) is usually a Right with an ErrorDTO object inside.

**Review by mastro993 — CHANGES_REQUESTED** [review]

**Pull request merged**